plugins/com.python.pydev.analysis/plugin.xml

```diff
@@ -177,7 +177,7 @@
                menubarPath="org.python.pydev.ui.actions.menu/pydev"
                tooltip="Force a code analysis for the selected elements (recursive)">
          </action>
-         <visibility><objectState name="projectNature" value="org.python.pydev.pythonNature"/></visibility>
+         <visibility><objectState name="contentTypeId" value="org.python.pydev.pythonfile"/></visibility>
```

Comment:
I experimented with this a bit, the existing objectState check has no sufficient test for something like "is a python file or a folder containing some python file". I've also experimented with changing the object class to PythonSourceFolder to enable it for python source folders (in addition to python files), but that does not seem to work very well for source folders added to a pydev project manually. At least the contribution does not show up there - but it does show up with a right-click on the project itself.

The last option I see (but haven't tried) is converting these actions to use the org.eclipse.ui.menus extension point for contribution. With that one could use command expressions to define the enablement/visibility and that supports a way to have code executed for such tests through propertyTester's.

Thats of course a bigger change so maybe not worth it for a corner case like ours...